mitodl/cookiecutter-djangoapp
# Many updates

Lots of updates to the cookiecutter:
- https://github.com/mitodl/mitxpro/pull/1538 - don't cache hash.txt
- https://github.com/mitodl/mitxpro/pull/681 - remove robots.txt, replace with django-robots
- Fix environment variable `UWSGI_THREADS` leftover from #168 
- Remove flow, except some dependencies required by `eslint-config-mitodl`
- Add `SITE_ID` environment variable
- Lots of package upgrades in package.json
- Relax deprecation warning errors
- Upgrade from raven to sentry -- fixes #152
- Upgrade from Python 3.6 to 3.8 -- fixes #161
- JS tweaks to match what we have in xpro
 - Update envs similar to xpro -- fixes #159 
 - Add generate_app_json -- https://github.com/mitodl/mitxpro/pull/360
 - Make `SECRET_KEY` required -- fixes #109 
 - Add `uwsgi_ignore_client_abort` setting -- fixes #153 
 - Fixed hot reloading (not sure if this is currently broken but I confirmed it works in this PR) - fixed #71 
 - Fix static file handling on local instances - fixes #143 
 - Tweaks to webpack config -- fixes #157 
 - Updates jsdom -- fixes #155 

#### Manual testing
I added `test_live_instance.sh` to create and set up a test server in a temp directory so you can verify that it works. If you run that command you should first see a line with a temporary directory -- that's where the cookiecutter was deployed. Eventually you see log output from `docker-compose up`. At that point, go to http://localhost:8073 and verify that you see a basic hello world message.

You can test the hot reloading by updating the border color in `layout.scss` in the temporary directory (not in the cookiecutter). You should see a near instant change in the browser confirming that the hot reloading worked. You can also update `App.js` and that should also hot reload with your text changes.
